pkg/controller/master/rancher/register.go

```diff
@@ -143,22 +152,35 @@ func (h *Handler) registerExposeService() error {
 		}
 		if enabled && vip.ServiceType == corev1.ServiceTypeLoadBalancer {
 			svc.Spec.Type = vip.ServiceType
-			svc.Spec.LoadBalancerIP = vip.LoadBalancerIP
+			// After kube-vip v0.5.2, it uses annotation kube-vip.io/loadbalancerIPs to set the loadBalancerIP
 			if strings.ToLower(vip.Mode) == vipDHCPMode {
-				svc.Annotations = map[string]string{
-					"kube-vip.io/requestedIP": vip.IP,
-					"kube-vip.io/hwaddr":      vip.HwAddress,
-				}
-				svc.Spec.LoadBalancerIP = vipDHCPLoadBalancerIP
+				svc.Annotations[keyKubevipRequestIP] = vip.IP
+				svc.Annotations[keyKubevipHwaddr] = vip.HwAddress
+				svc.Annotations[keyKubevipLoadBalancerIPs] = vipDHCPLoadBalancerIP
 			} else {
-				svc.Spec.LoadBalancerIP = vip.IP
+				svc.Annotations[keyKubevipLoadBalancerIPs] = vip.IP
 			}
 		}
 
 		if _, err := h.Services.Create(svc); err != nil {
 			return err
 		}
 	}
+	// Update annotations of the ingress-expose service created before Harvester v1.2.0
+	if svc.Annotations == nil || svc.Annotations != nil && svc.Annotations[keyKubevipIgnoreServiceSecurity] != trueStr {
```

Comment:
you should be able to check it directly via 
`if svc.Annotations == nil || svc.Annotations[keyKubevipIgnoreServiceSecurity] != trueStr `,
also, please help to address the CodeFactor warning.